src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -159,25 +158,40 @@ object BlockVTK {
     }
     offsetIterator(offsets, localOffsets.toSeq)
   }
+
+  /*
+   * FIXME: This is very inefficient and we probably want to use the right data
+   * structures to deal with duplicates for us.
+   */
+  private def distinctArrays(vertices: Seq[Array[Double]], previous: Seq[Array[Double]]=Nil): Seq[Array[Double]] = {
+    vertices match {
+      case Nil => previous
+      case vert+:verts =>
+        if (previous.exists(previousVertex => previousVertex sameElements vert)) {
+          distinctArrays(verts, previous)
+        } else {
+          distinctArrays(verts, vert+:previous)
+        }
+    }
+  }
 }
 
 /**
   * Simple data structure to contain data that represents a rock block that can be turned into vtk format
-  * @param Rock block 
   * @constructor Create a new rock block in format that can be turned into vtk by rockProcessor
   */
 case class BlockVTK(block: Block) {
   val orientedVertices = BlockVTK.orientVertices(block)
-  val tupleVertices = orientedVertices.values.flatten.toSeq.distinct
-  val connectivityMap = BlockVTK.connectivity(orientedVertices, tupleVertices)
-  val vertices = tupleVertices flatMap { t =>
-    List(t._1, t._2, t._3)
+  val arrayVertices = BlockVTK.distinctArrays(orientedVertices.values.flatten.toSeq)
```

Comment:
There are two problems to deal with here. One is the cost of removing duplicates, and the other is avoiding boxing.

You're right that using `toSeq` creates `WrappedArray`s, which to my understanding have to box the `Double`s. So that's why I didn't want to use the built in `distinct`.

My implementation here is a simple but inefficient O(n^2) algorithm. It turns out that Scala will use a HashSet underneath to make `distinct` O(n). See here if you're interested: https://lampsvn.epfl.ch/trac/scala/browser/scala/tags/R_2_9_1_final/src//library/scala/collection/SeqLike.scala

So we need to either use the fast approach that will take more memory or use a slower approach that avoids boxing. Any thoughts here? How often will we need to perform this operation, and what are the typical sizes of the arrays involved?